src/main/kotlin/org/veupathdb/lib/ldap/NetDesc.kt

```diff
@@ -0,0 +1,12 @@
+package org.veupathdb.lib.ldap
+
+enum class Platform {
+    ORACLE, POSTGRES
+}
+
+data class NetDesc(
+    val host: String,
+    val port: UShort,
+    val identifier: String,
+    val platform: Platform
+)
```

Comment:
It seems like it might be cleaner to make this an interface that both `PostgresNetDesc` and `OracleNetDesc` implement?  Then we don't need to do extra type conversions to `NetDesc`.

The `identifier` property on each implementation could just be an alias of `serviceName` or `dbname`.